node/src/components/small_network/message.rs

```diff
@@ -432,16 +434,22 @@ mod tests {
 
     /// Serialize a message using the standard serialization method for handshakes.
     fn serialize_message<M: Serialize>(msg: &M) -> Vec<u8> {
-        // The actual serialization/deserialization code can be found at
-        // https://github.com/carllerche/tokio-serde/blob/f3c3d69ce049437973468118c9d01b46e0b1ade5/src/lib.rs#L426-L450
+        let mut serializer = MessagePackFormat;
 
-        rmp_serde::to_vec(&msg).expect("handshake serialization failed")
+        Pin::new(&mut serializer)
+            .serialize(&msg)
+            .expect("handshake serialization failed")
```

Comment:
Shouldn't we return a `Result` here instead? I understand that most likely we can't recover from it but we could try to handle it "gracefully".

Reply:
That would be true usually, but this is testing code. The actual code has an error for this, `ConnectionError::CouldNotEncodeOurHandshake`.

Reply:
I missed that it's a test module. I looked at the path only.